olive/model/handler/onnx.py

```diff
@@ -74,29 +74,58 @@ def prepare_session(
         execution_providers: Union[str, List[str]] = None,
         rank: Optional[int] = None,
     ):
+        import onnxruntime as ort
+
         # user provided inference_settings > model's inference_settings > default settings
         inference_settings = inference_settings or self.inference_settings or {}
         # deep copy to avoid modifying the original settings
         inference_settings = deepcopy(inference_settings)
 
         # if user doesn't not provide ep list, use default value([ep]). Otherwise, use the user's ep list
-        # user provided ep list > eps given by arguments > default eps
-        execution_providers = inference_settings.get("execution_provider") or execution_providers
-        if not execution_providers:
-            execution_providers = self.get_default_execution_providers(device)
-        elif isinstance(execution_providers, str):
+        # eps given by arguments > user provided ep list > default eps
+        # Note by myguo: in DLIS scenarios, the inferenece_settings is something like:
+        #  {'execution_provider': [('MIGraphXExecutionProvider', {})]}
+        # and the provider_lists specified in perf_tuning is: ['ROCMExecutionProvider', 'MIGraphXExecutionProvider'].
+        # Originally, the inference_settings priority is higher than the execution_providers specified in arguments.
+        # as the result, when calling OnnxEvaluator.evaluate with the provider_list in baseline evaluation,
+        # the execution provider will be used as ["MIGraphXExecutionProvider"].
+        # later, the check_ort_fallback(session, execution_providers) will check the
+        # whether the underlying session's providers(MIGraphXExecutionProvider, CPUExecutionProvider) contains the
+        # execution_providers(ROCMExecutionProvider, MIGraphXExecutionProvider). In this case, the
+        # ROCMExecutionProvider is excluded when creating inference session. and the OliveEvaluationError exception
+        # will be raised.
+        # To fix this issue, we need to make sure the execution_providers specified in arguments has higher priority
+        if execution_providers:
```

Comment:
Olive evaluator always provides execution provider https://github.com/microsoft/Olive/blob/3a774a75592cd4d12c5836dea19d6d462ccbaeac/olive/systems/local.py#L49 

So, wouldn't this ignore the perf tuned inference settings when evaluating the output model of the perf-tuning pass?

Reply:
Or it doesn't matter since only the `provider_options` is ignored in this case and we don't tune it. But what about settings like `trt_fp16_enable` and `enable_cuda_graph`? https://github.com/microsoft/Olive/blob/3a774a75592cd4d12c5836dea19d6d462ccbaeac/olive/passes/onnx/perf_tuning.py#L135 

Perhaps, it would be more complete if we resolve the `ep, provider_options` for both the user provided `ep` and model's inference settings. As use the inference setting's provider options for eps from user that don't have their own provider options?

Reply:
fixed.